src/video/emscripten/SDL_emscriptenevents.c

```diff
@@ -488,7 +498,7 @@ Emscripten_HandleKeyPress(int eventType, const EmscriptenKeyboardEvent *keyEvent
     char text[5];
     Emscripten_ConvertUTF32toUTF8(keyEvent->charCode, text);
     SDL_SendKeyboardText(text);
-    return 1;
+    return SDL_GetEventState(SDL_TEXTINPUT) == SDL_ENABLE;
```

Comment:
I think we should test this first. Text input can be enabled and disabled, and is disabled by default. Maybe the whole content of this function should be put into a [`if (SDL_IsTextInputActive())`](https://wiki.libsdl.org/SDL_IsTextInputActive) because normally (outside of emscripten), SDL_TEXTINPUT events are only created when [`SDL_StartTextInput`](https://wiki.libsdl.org/SDL_StartTextInput) was called before.

Also, this could be a problem if the browsers default handling of keypress does the same as the handling of keydown for space / tab.